File: Models/Event.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Runtime.Serialization;
using System.Text.Json.Serialization;

namespace ASP_MVC_ANGULAR.Models
{
    [Table("event")]
    public class Event
    {
        [Column("event_id")]
        public int EventId { get; set; }

        [Column("event_title", TypeName = "varchar(180)")]
        [Required]
        public string Title { get; set; }

        [Column("event_description")]
        [Required]
        public string Description { get; set; }

        [Column("event_date")]
        [Required]
        public DateTime EventDate { get; set; }

        [Column("category_id")]
        public int CategoryId { get; set; }

        public Category Category { get; set; }

        [JsonIgnore]
        public ICollection<Participate> Participates { get; set; }
    }
}

File: ClientApp/src/app/shared/models/event.ts
import {User} from './user';
import {Category} from './category';

export class Event {
  public eventId: number;
  public title: string;
  public description: string;
  public eventDate: string;
  public categoryId: number;
  public category: Category;
  public users: User[];

  public constructor(init?: Partial<Event>) {
    Object.assign(this, init);
  }

  deserialize(input: any): Event {
    Object.assign(this, input);
    // this.users = input.users.map(user => new User().deserialize(user));
    return this;
  }
}
